aws/resource_aws_wafv2_regex_pattern_set_test.go

```diff
@@ -0,0 +1,389 @@
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSWafv2RegexPatternSetExists(resourceName, &v),
+					testAccMatchResourceAttrRegionalARN(resourceName, "arn", "wafv2", regexp.MustCompile(`regional/regexpatternset/.+$`)),
+					resource.TestCheckResourceAttr(resourceName, "name", regexPatternSetName),
+					resource.TestCheckResourceAttr(resourceName, "description", regexPatternSetName),
+					resource.TestCheckResourceAttr(resourceName, "scope", wafv2.ScopeRegional),
+					resource.TestCheckResourceAttr(resourceName, "regular_expression_list.#", "2"),
+					resource.TestCheckResourceAttr(resourceName, "regular_expression_list.1641128102.regex_string", "one"),
+					resource.TestCheckResourceAttr(resourceName, "regular_expression_list.265355341.regex_string", "two"),
+					resource.TestCheckResourceAttr(resourceName, "tags.%", "2"),
+					resource.TestCheckResourceAttr(resourceName, "tags.Tag1", "Value1"),
+					resource.TestCheckResourceAttr(resourceName, "tags.Tag2", "Value2"),
+				),
+			},
+			{
+				Config: testAccAwsWafv2RegexPatternSetConfig_Update(regexPatternSetName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSWafv2RegexPatternSetExists(resourceName, &v),
+					testAccMatchResourceAttrRegionalARN(resourceName, "arn", "wafv2", regexp.MustCompile(`regional/regexpatternset/.+$`)),
+					resource.TestCheckResourceAttr(resourceName, "name", regexPatternSetName),
+					resource.TestCheckResourceAttr(resourceName, "description", "Updated"),
+					resource.TestCheckResourceAttr(resourceName, "scope", wafv2.ScopeRegional),
+					resource.TestCheckResourceAttr(resourceName, "regular_expression_list.#", "3"),
+					resource.TestCheckResourceAttr(resourceName, "regular_expression_list.1641128102.regex_string", "one"),
+					resource.TestCheckResourceAttr(resourceName, "regular_expression_list.265355341.regex_string", "two"),
+					resource.TestCheckResourceAttr(resourceName, "regular_expression_list.2339296779.regex_string", "three"),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+				ImportStateIdFunc: testAccAWSWafv2RegexPatternSetImportStateIdFunc(resourceName),
+			},
+		},
+	})
+}
+
+func TestAccAwsWafv2RegexPatternSet_Disappears(t *testing.T) {
+	var v wafv2.RegexPatternSet
+	regexPatternSetName := fmt.Sprintf("regex-pattern-set-%s", acctest.RandString(5))
+	resourceName := "aws_wafv2_regex_pattern_set.test"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSWafv2RegexPatternSetDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAwsWafv2RegexPatternSetConfig_Minimal(regexPatternSetName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSWafv2RegexPatternSetExists(resourceName, &v),
+					testAccCheckAWSWafv2RegexPatternSetDisappears(resourceName),
```

Comment:
similar suggestion made in #12119, to use the generic method now available in the provider https://github.com/terraform-providers/terraform-provider-aws/blob/07f8ab992c780c4911832234afbbf67feea2fe66/aws/provider_test.go#L590 in place of a custom one